### PR TITLE
PAR-988: Fixed admin issues adding entities.

### PR DESCRIPTION
**Pull request by kalpaitch**

### CODE REVIEW

File: web/modules/custom/par_data/src/Controller/ParDataAddController.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+   * @return array
+   *   A render array for a list of the trance bundles/types that can be added
+   *   or if there is only one type/bunlde defined for the site, the function
+   *   returns the add page for that bundle/type.
+   */
+  public function add(Request $request) {
+    $entity_type = $this->storage->getEntityType()->id();
+    $bundle_entity_type = $this->storage->getEntityType()->getBundleEntityType();
+    $types = $this->typeStorage->loadMultiple();
+    if ($types && count($types) == 1) {
```
Comment (stemount) [inline web/modules/custom/par_data/src/Controller/ParDataAddController.php]: can we type match here as the lines below? e.g. `(count($types) === 0)`

File: web/modules/custom/par_data/src/Controller/ParDataAddController.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      $type = key($types);
+      return $this->addForm($type, $request);
+    }
+    if (count($types) === 0) {
+      return [
+        '#markup' => $this->t('You have not created any %bundle types yet. @link to add a new type.', [
+          '%bundle' => $entity_type,
+          '@link' => $this->l($this->t('Go to the type creation page'), Url::fromRoute('entity.' . $bundle_entity_type . '.add_form')),
```
Comment (stemount) [inline web/modules/custom/par_data/src/Controller/ParDataAddController.php]: how about `"entity.{$bundle_entity_type}.add_form"` (is this a coding standard?)

Comment (kalpaitch) [reply web/modules/custom/par_data/src/Controller/ParDataAddController.php]: Well it's how we define our routes in the ParDataRoutes plugin.

File: web/modules/custom/par_data/src/Controller/ParDataAddController.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Url;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class TranceAddController.
```
Comment (jon-edery) [inline web/modules/custom/par_data/src/Controller/ParDataAddController.php]: Comment update...

File: web/modules/custom/par_data/src/Controller/ParDataAddController.php [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    }
+    if (count($types) === 0) {
+      return [
+        '#markup' => $this->t('You have not created any %bundle types yet. @link to add a new type.', [
+          '%bundle' => $entity_type,
+          '@link' => $this->l($this->t('Go to the type creation page'), Url::fromRoute('entity.' . $bundle_entity_type . '.add_form')),
+        ]),
+      ];
+    }
+    return ['#theme' => 'trance_content_add_list', '#content' => $types];
```
Comment (jon-edery) [inline web/modules/custom/par_data/src/Controller/ParDataAddController.php]: Is there no Par standard for theming this?

File: web/modules/custom/par_data/src/Controller/ParDataAddController.php [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+  /**
+   * Presents the creation form for trance entities of given bundle/type.
+   *
+   * @param EntityInterface $par_data_type
```
Comment (jon-edery) [inline web/modules/custom/par_data/src/Controller/ParDataAddController.php]: $par_data_entity_type

**Comment by jon-edery** [comment]
Minor updates looks good.

File: web/modules/custom/par_data/src/Controller/ParDataAddController.php (hunk 4 above)
Comment (kalpaitch) [reply web/modules/custom/par_data/src/Controller/ParDataAddController.php]: For reference I wrote none of this code, it's basically all the stuff Adrian wrote for Eurostar #sharenoblame

Comment (kalpaitch) [reply web/modules/custom/par_data/src/Controller/ParDataAddController.php]: Well it's only a list, let's just keep using all that Trance stuff for now.

**Pull request merged**